src/util/timestamp.js

```diff
@@ -66,3 +68,18 @@ export function validate(timestampMs: number): TimestampMs {
   }
   return new Date(asNumber).valueOf();
 }
+
+export function validateTimestampISO(timestampISO: string): TimestampISO {
+  // Verify that it can be converted without throwing an error
+  fromISO(timestampISO);
+  return timestampISO;
+}
+
+export const timestampMsParser: C.Parser<TimestampMs> = C.fmap(
+  C.number,
+  validateTimestampMs
+);
+export const timestampISOParser: C.Parser<TimestampISO> = C.fmap(
+  C.string,
+  validateTimestampISO
+);
```

Comment:
Did something similar in initiatives parser:
```
const TimestampParser = C.fmap(C.string, (t) => toISO(fromISO(t)));
```

Not sure if there's any practical benefit in forcing it to go through our `toISO` function, but thought I'd share.

Reply:
That's a neat trick, could have made the implementation for https://github.com/sourcecred/sourcecred/pull/2226 a bit more concise. WOuldn't have needed to factor out the validator function.